doc/usage/restructuredtext/basics.rst

```diff
@@ -221,6 +221,12 @@ You can also separate the link and the target definition (:duref:`ref
 
    .. _a link: https://domain.invalid/
 
+If you want to set a custom target name::
+
+    This paragraph contains `a link with a custom target name <custom_>`_.
+
+    .. _custom: https://domain.invalid/
```

Comment:
What do you mean by 'custom target name'? This is called an 'alias' in the Docutils documentation: https://www.docutils.org/docs/ref/rst/restructuredtext.html#embedded-uris-and-aliases

Reply:
I wrote it that way based on the preceding paragraph: 

> You can also separate the link and the target definition [...]

I wasn't aware of the alias nomenclature - to be honest, largely because I searched "restructured text link syntax" or something like that, and this page was the top result. As someone pretty unfamiliar with the rST world I honestly _kind of_ thought, without having really consciously thought about it, that this was the canonical format documentation. So it didn't even occur to me to check elsewhere.

Knowing this now, I can update the patch (and probably update the preceding sentence at the same time). But I can also see an argument for just saying, this syntax is a little advanced, here's a link to upstream and no example?

Reply:
I think better to provide an example, especially given the PR has already been created.

Reply:
Thanks @AA-Turner, I now tried multiple `DOI link`__ and this has indeed solved my problem!

I think I had used the syntax before but that this was at a time when I was not aware of the `WARNING: Duplicate explicit target name` one gets when one uses `` `DOI link <https://.../>`_ `` multiple times with distinct URLs.  By the way I realize now that the `WARNING` is emitted only if the URL changes.  No `WARNING` if one uses multiple times same URL with same text at various locations.

Reply:
I realize now in my reST source I had also some `` `FOO <A_>`_ `` and `` `FOO <B_>`_ `` syntax with same `FOO` and then `<A_>` is resolved by a `.. _A: URL` and `<B_>` by a `.. _B: OTHERURL` and there was no Docutils warning then.  So I was pretty much surprised by the warnings caused by `` `DOI <URL>`_ ``  and `` `DOI <OTHERURL>`_ `` which I noticed later.  I recall now that I thought about using the same I had earlier in the file with `A` versus `B` sharing the same `FOO`, but I had other priorities and forgot about it afterwards.

Reply:
@jfbu this is a code review thread. Is there an action item here for me?